docs/architecture/legacy-module-integration.md

```diff
@@ -0,0 +1,239 @@
+# 🔙 Legacy Module Integration
+
+This document outlines how existing legacy code is integrated into the new modular architecture of the application and
+the strategy for its migration. The core principle is to isolate legacy code and provide a controlled way for
+newer modules to interact with legacy functionality without becoming directly dependent on it.
+
+> **Note:** This document should be read in conjunction with [Module Structure](module-structure.md) and [Module Organization](module-organization.md) to get a complete understanding of the modular architecture.
```

Comment:
You can use the fancy note blocks from https://github.com/orgs/community/discussions/16925

Reply:
There is support for mdbook, will add it.